### PR TITLE
fix: move the vote fetching out of the individual cards

### PR DESCRIPTION
**Pull request by jasonericdavis**
## What type of PR is this? (check all applicable)

- [x] ♻️ Refactor
- [ ] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 👷 Optimization
- [ ] 📝 Documentation Update
- [ ] 🔖 Release
- [ ] 🚩 Other

## Description

Currently fetching the votes for each repository is handled by the invidual repository cards in src/components/PostGrid.jsx which results in multiple http request being made when loading the entire grid. This PR is a suggestion to add the votes to the query when the repository data is being fetched. From there the invidual repository cards can handle updates to the votes. 

I am making this PR a draft because I wanted to get others' thoughts on it.

This PR is a suggested update to the voting logic in PR #67  

## Related Tickets & Documents

- Update to PR #67  that implemented Issue #12 

## Mobile & Desktop Screenshots/Recordings

This PR does not include any visual changes and therefore a screenshot is not required.

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 readme
- [ ] 📜 contributing.md
- [ ] 📓 docs
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

## [optional] Are there any post-deployment tasks we need to perform?
No post-deployment tasks are needed

## [optional] What gif best describes this PR or how it makes you feel?
What if

### CODE REVIEW

**Comment by 0-vortex** [comment]
Super excited for this! 🍕

File: src/lib/database.js [hunk 1]
```diff
@@ -25,3 +25,15 @@ export async function updateVotesByRepo(repoName, votes) {
...
+export async function fetchRecommendations() {
+  const { data: recommendations, error } = await supabase
+    .from('recommendations')
+    .select('repo_name, description,stars,issues, total_stars, avg_recency_score, contributors, votes')
+    .limit(25)
+    .order('total_stars', { ascending: false });
+
+  console.error(error);
+
+  return recommendations;
```
Comment (bdougie) [inline src/lib/database.js:38]: This will be useful for pagination/infinite scroll.

File: src/components/PostsWrap.jsx [hunk 2]
```diff
@@ -1,16 +1,23 @@
...
   const [isGrid, setIsGrid] = useState(true);
   const [activeLink, setActiveLink] = useState('popular');
-  const fetchedData = [...postData];
+  const [fetchedData, setFetchedData] = useState([]);
+  
+  useEffect(() => {
+    fetchRecommendations().then((data) => {
+      setFetchedData(data);
+    });
+  }, []);
```
Comment (bdougie) [inline src/components/PostsWrap.jsx:20]: This is a welcomed change and makes a ton more sense.

**Review by bdougie — APPROVED** [review]

File: src/lib/database.js [hunk 3]
```diff
@@ -25,3 +25,15 @@ export async function updateVotesByRepo(repoName, votes) {
 
   return recommendations[0].votes;
 }
+
+export async function fetchRecommendations() {
+  const { data: recommendations, error } = await supabase
+    .from('recommendations')
+    .select('repo_name, description,stars,issues, total_stars, avg_recency_score, contributors, votes')
```
Comment (bdougie) [inline src/lib/database.js:32]: This is the first step ni resolving #38

**Comment by bdougie** [comment]
https://github.com/open-sauced/hot/commit/1c23f30611d94320d4265170cb6c7832dc775f45

**Pull request closed without merging**